server/src/main/java/org/elasticsearch/transport/Transport.java

```diff
@@ -118,12 +119,19 @@ void sendRequest(long requestId, String action, TransportRequest request, Transp
         boolean isClosed();
 
         /**
-         * Returns the version of the node this connection was established with.
+         * Returns the version of the node on the other side of this channel.
          */
         default Version getVersion() {
```

Comment:
This is left as-is to distinguish between the node version you're connecting to & transport version used to communicate on the channel